### PR TITLE
1777 validate amendment number

### PR DESCRIPTION
**Pull request by marcellmueller**
Implements #1777 

Also updated some change request form styles. Please check this closely for bugs, especially in the `handleSave` section there was a lot more logic I had to add than expected 🙇‍♂️

### CODE REVIEW

**Review by rafasdc — COMMENTED** [review]
When you add a new amendment and then edit the number, the UI will temporarily show correct as an update but a new one gets created. For example, if I create amendment 19, then realized it should have been 18, go to edit and change to 18 it will show ok on the UI but after a refresh both 18 and 19 will be visible and exist on the DB.

**Comment by marcellmueller** [comment]
> When you add a new amendment and then edit the number, the UI will temporarily show correct as an update but a new one gets created. For example, if I create amendment 19, then realized it should have been 18, go to edit and change to 18 it will show ok on the UI but after a refresh both 18 and 19 will be visible and exist on the DB.

Damn, nice catch. I updated the mutation to accept the old rowId of the previous change request and use that to archive instead of the amendment. 

Only small caveat I noticed is that if you edit a change request to a new amendment number, then try and reuse the old one it says it is taken until you refresh the page. Though I kind of think that could be it's own ticket if we see it as an issue.

**Comment by marcellmueller** [comment]
@rafasdc can you check this out again when you have time, I think the edit change request amendment number issue is taken care of thanks.

**Comment by rafasdc** [comment]
> [ ](/marcellmueller)

The edit is still being weird for me, I can reuse a previous amendment number on the same screen before refreshing, seems the UI is updating properly but the DB is still not archiving or updating the edited amendment. 

image

**Comment by marcellmueller** [comment]
@rafasdc ah sorry about that 😓  I guess I had too many change requests and didn't notice. It was a couple issues, I needed to use the `rowId` for the change request plus it was using the wrong `old_change_request_id` in the postgres mutation itself.

File: app/components/Analyst/Project/ProjectForm.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ const ToggleRight = styled.div`
   align-items: center;
   margin-left: auto;
   margin-top: auto;
-  margin-bottom: auto;
+  margin-bottom: autoo
```
Comment (rafasdc) [inline app/components/Analyst/Project/ProjectForm.tsx:17]: Possible typo? auto;

Comment (marcellmueller) [reply app/components/Analyst/Project/ProjectForm.tsx]: Fixed!

File: app/components/Analyst/Project/ProjectInformation/ProjectInformationForm.tsx [hunk 2]
```diff
@@ -191,29 +223,57 @@
       } else if (isSowUploaded) {
         delete newFormData?.isSowUploadError;
       }
-
       if (isChangeRequest) {
+        console.log('old change request id', currentChangeRequestData?.rowId);
```
[the remark this answers is not in the document]
Comment (marcellmueller) [reply app/components/Analyst/Project/ProjectInformation/ProjectInformationForm.tsx]: Fixed!

**Review by rafasdc — APPROVED** [review]
Working great now, awesome work! 🥂

**Pull request merged**